server/ladder_service.py

```diff
@@ -125,8 +126,29 @@ async def fetch_map_pools(self, conn) -> Dict[int, Tuple[str, List[Map]]]:
             _, map_list = map_pool_maps[id_]
             if row.map_id is not None:
                 map_list.append(
-                    Map(row.map_id, row.display_name, row.filename)
+                    Map(row.map_id, row.display_name, row.filename, row.weight)
                 )
+            elif row.map_params is not None:
+                try:
+                    params = json.loads(row.map_params)
+                    map_type = params["type"]
+                    if map_type == "neroxis":
+                        map_list.append(NeroxisGeneratedMap.of(params=params, weight=row.weight))
```

Comment:
keyword arguments really necessary here?
```suggestion
                        map_list.append(NeroxisGeneratedMap.of(params, row.weight))
```